graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -249,4 +250,63 @@ static boolean hasExpectedSchema(Table table) {
     Column contents = table.column(CONTENTS_COLUMN_NAME);
     return contents != null && contents.type() == Column.Type.Varchar;
   }
+
+  public Map<String, Object> tryDeployNewSchema(
+      String namespace, Callable<Map<String, Object>> newSchemaCreator) throws Exception {
+    ensureTableExists(namespace);
+    if (transitionToSchemaDeploymentInProgress(namespace)) {
+      try {
+        return newSchemaCreator.call();
+      } finally {
+        transitionToNoSchemaDeploymentInProgress(namespace);
+      }
+    } else {
+      throw new IllegalStateException(
+          "There is a pending schema deployment process. Please retry later.");
+    }
+  }
+  /**
+   * Returns true if there was no schema deployment in progress. It means that that is safe to
+   * deploy a new schema. If it returns false, it means that there is already schema deployment in
+   * progress.
+   */
+  private boolean transitionToSchemaDeploymentInProgress(String namespace)
+      throws ExecutionException, InterruptedException {
+
+    BuiltCondition schemaDeploymentInProgressNull =
+        BuiltCondition.of(DEPLOYMENT_IN_PROGRESS, Predicate.EQ, null);
+
+    BoundQuery updateDeploymentToInProgress =
+        dataStore
+            .queryBuilder()
+            .update(namespace, TABLE_NAME)
+            .value(DEPLOYMENT_IN_PROGRESS, true)
+            .where(KEY_CONDITION)
+            .ifs(schemaDeploymentInProgressNull)
+            .build()
+            .bind();
+
+    ResultSet resultSet = dataStore.execute(updateDeploymentToInProgress).get();
+    Row row = resultSet.one();
+    return row.getBoolean(APPLIED_COLUMN_NAME);
+  }
+
+  private void transitionToNoSchemaDeploymentInProgress(String namespace)
+      throws ExecutionException, InterruptedException {
+    BuiltCondition schemaDeploymentInProgressTrue =
+        BuiltCondition.of(DEPLOYMENT_IN_PROGRESS, Predicate.EQ, true);
+
+    BoundQuery updateDeploymentToNotInProgress =
+        dataStore
+            .queryBuilder()
+            .update(namespace, TABLE_NAME)
+            .value(DEPLOYMENT_IN_PROGRESS, null)
+            .where(KEY_CONDITION)
+            .ifs(schemaDeploymentInProgressTrue)
```

Comment:
We don't need an LWT here: if only one user is allowed to set the flag, then it's guaranteed that no more than one user will try to unset it.